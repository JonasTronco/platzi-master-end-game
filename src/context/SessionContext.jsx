import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
