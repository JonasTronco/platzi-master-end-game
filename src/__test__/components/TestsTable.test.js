import TestsTable from '../../components/TestsTable';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import { SessionProvider } from '../../context/SessionContext';
describe('TestsTable Component', () => {
  const obj = {
    session: {
      token: '',
      user: {
        id: '',
        username: '',
        typeOfUser: '',
        isActive: true,
        imageURL: '',
        firstName: '',
        lastName: '',
        defaultPath: ''
      }
    }
  };
  test('Should render TestTable component', () => {
    const component = create(
      <SessionProvider value={obj}>
        <TestsTable username='' />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('<TestsTable />', () => {
    const component = mount(
      <SessionProvider value={obj}>
        <TestsTable username='' />
      </SessionProvider>
    );
    expect(component.length).toEqual(1);
  });
});
