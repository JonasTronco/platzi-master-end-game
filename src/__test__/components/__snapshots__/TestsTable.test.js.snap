// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestsTable Component Should render TestTable component 1`] = `""`;
